src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -0,0 +1,208 @@
+package io.github.oliviercailloux.y2018.apartments.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.github.oliviercailloux.y2018.apartments.apartment.Apartment;
+
+import javax.json.bind.JsonbBuilder;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.ArrayList;
+import static com.google.common.base.Preconditions.checkArgument;
+
+import javax.json.bind.Jsonb;
+
+/**
+ * The Class JsonConvert contains all function to transform Apartment object to
+ * JSON and vice-versa.
+ * 
+ * @author Etienne CARTIER & Morgane FIOT
+ */
+public abstract class JsonConvert {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(JsonConvert.class);
+
+	/**
+	 * The method return the default path to JSON file.
+	 * 
+	 * @return <i>Path</i> where to read and write JSON.
+	 */
+	private static final Path apartmentPathJson() {
+		Path path = Paths.get("Apartment_Json.json");
+		return path;
+	}
+
+	/**
+	 * The method return a default JSON file read by jsonToApartments.
+	 * 
+	 * @return <i>Path</i> where jsonToApartments will read.
+	 */
+	private static final Path startApartment() {
+		Path path = Paths
+				.get("src/main/resources/io/github/oliviercailloux/y2018/jsonResources/defaultJsonToApartments.json");
+		return path;
+	}
+
+	/**
+	 * Converts an Apartment object to a JSON file with the default path
+	 * APARTMENT_PATH_JSON.
+	 *
+	 * @param a <i>Apartment</i> object to convert into JSON
+	 * @throws IOException if the JSON file can't be created.
+	 */
+	public static void apartmentToJson(Apartment a) throws IOException {
+		apartmentToJson(a, apartmentPathJson());
+	}
+
+	/**
+	 * Converts an Apartment object to a JSON file.
+	 *
+	 * @param a        <i>Apartment</i> object to convert into JSON
+	 * @param jsonPath <i>Path</i> the path where to create the JSON file
+	 * @throws IOException if the JSON file can't be created.
+	 */
+	public static void apartmentToJson(Apartment a, Path jsonPath) throws IOException {
+		Jsonb jsonb = JsonbBuilder.create();
+
+		try (BufferedWriter writer = Files.newBufferedWriter(jsonPath, StandardCharsets.UTF_8)) {
+			writer.write(jsonb.toJson(a));
+			writer.close();
+
+			LOGGER.info("Apartment have been converted with success");
+		}
+	}
+
+	/**
+	 * Read a JSON file which contains an Apartment.
+	 *
+	 * @param jsonPath <i>Path</i> the path where the JSON file is located
+	 * @return <i>String</i> containing an Apartment into JSON format
+	 * @throws IOException if the file can't be convert into JSON format.
+	 */
+	protected static String readApartmentFromJson(Path jsonPath) throws IOException {
+		try (BufferedReader reader = Files.newBufferedReader(jsonPath, StandardCharsets.UTF_8)) {
```

Comment:
Utiliser Files.read. Généralement, se demander si une fct n’existe pas déjà avant de la re-coder.

Reply:
Fix in commit d09c20788b385776f2f08503bee5a1fe936631da